esp-wifi/src/ble/btdm.rs

```diff
@@ -178,10 +169,13 @@ unsafe extern "C" fn mutex_unlock(_mutex: *const ()) -> i32 {
 }
 
 unsafe extern "C" fn queue_create(len: u32, item_size: u32) -> *const () {
-    if len != 5 && item_size != 8 {
-        panic!("Unexpected queue spec {} {}", len, item_size);
+    let raw_queue = RawQueue::new(len as usize, item_size as usize);
+    let ptr = unsafe { crate::compat::malloc::malloc(size_of_val(&raw_queue)) as *mut RawQueue };
```

Comment:
uh... can we `Box::leak(Box::new())` instead? Also, `queue_delete` is unimplemented so nothing frees this right now. Are we leaking memory?

Reply:
would be something I'm happy to do with allocator-api but not with the global allocator

`queue_delete` was never called for me - just to be safe I'll implement it